cmake/FindTut.cmake

```diff
@@ -0,0 +1,47 @@
+################################################################################
+#
+# \file      cmake/FindTut.cmake
+# \author    C. Junghans
+# \copyright 2016, Los Alamos National Security, LLC.
+# \brief     Find the Tut header
+# \date      Mon 05 Dec 2016 14:21:53 AM MST
+#
+################################################################################
+
+# Find Tut headers and libraries
+#
+#  Tut_FOUND        - True if Tut is found
+#  Tut_INCLUDE_DIRS - Tut include files paths
+#
+#  Set Tut_ROOT before calling find_package to a path to add an additional
+#  search path, e.g.,
+#
+#  Usage:
+#
+#  set(Tut_ROOT "/path/to/custom/tut") # prefer over system
+#  find_package(Tut)
+#  if(Tut_FOUND)
+#    include_directories(${Tut_INCLUDE_DIR})
+#  endif()
+
+find_package(PkgConfig)
+
+# If already in cache, be silent
+if (Tut_INCLUDE_DIR)
+  set (Tut_FIND_QUIETLY TRUE)
+  pkg_check_modules(PC_TUT QUIET tut)
```

Comment:
Debian has an ancient (2007) version of tut. Is there a way to detect the TUT version so we can reject it if it is not recent enough? In particular, `tut/tut_runner.hpp` is missing in that old version, while the newer one (as well as in ours at `tpl/src/tut') has it.

If there is not good way of detecting the `tut` version, we could include a search for the header 'tut_runner.hpp'. What do you think?